app/layout.py

```diff
@@ -0,0 +1,82 @@
+"""Module for generating figure grid layout."""
+from dash import html  # type: ignore
+
+
+class GridBuilder:
+    """_summary_."""
+
+    def __init__(
+        self,
+        rows: int,
+        cols: int,
+        row_heights: list[str] = None,  # type: ignore
+        col_widths: list[str] = None,  # type:ignore
+    ):
+        """_summary_.
+
+        Args:
+            rows (int): _description_
+            cols (int): _description_
+            row_heights (list[str], optional): _description_. Defaults to None.
+            col_widths (list[str], optional): _description_. Defaults to None.
+        """
+        self.rows = rows
+        self.cols = cols
+        self.row_heights = row_heights
+        self.col_widths = col_widths
+        self.row_heights = row_heights
+        self.col_widths = col_widths
+        if not self.row_heights:
+            self.row_heights = [f"{100 / rows}vh"] * rows
+        if not self.col_widths:
+            self.col_widths = [f"{100 / cols}%" for _ in range(cols)]
+        self.grid_elements = [[None] * cols for _ in range(rows)]
```

Comment:
Double up of heights and widths assignment. Also I'd prefer to see the `self.` objects only assigned once, not assigned, then checked, then changed. Something like this (although maybe would be good to check the input values are valid):

```suggestion
        if row_heights is None:
            row_heights = [f"{100 / rows}vh"] * rows
        if col_widths is None:
            col_widths = [f"{100 / cols}%" for _ in range(cols)]
        
        self.rows = rows
        self.cols = cols
        self.row_heights = row_heights
        self.col_widths = col_widths
        self.grid_elements = [[None] * cols for _ in range(rows)]
```